File: Lab5.Web/src/infrastructure/repositories/InMemoryBookRepository.js
import { Book } from '../../domain/entities/Book.js';

export class InMemoryBookRepository {
  constructor() {
    this.books = [
      new Book('b1', 'Clean Code', 'Robert C. Martin', 2),
      new Book('b2', 'Refactoring', 'Martin Fowler', 1),
      new Book('b3', 'Design Patterns', 'Gang of Four', 3)
    ];
  }

  getAll() {
    return this.books;
  }

  getById(id) {
    return this.books.find((book) => book.id === id) ?? null;
  }
}

File: Lab5/Infrastructure/Repositories/InMemoryBookRepository.cs
using Lab5.Domain.Entities;
using Lab5.Domain.Interfaces;

namespace Lab5.Infrastructure.Repositories;

public sealed class InMemoryBookRepository : IBookRepository
{
    private readonly IReadOnlyCollection<Book> _books =
    [
        new Book(Guid.Parse("00000000-0000-0000-0000-000000000101"), "Clean Code", "Robert C. Martin", 2),
        new Book(Guid.Parse("00000000-0000-0000-0000-000000000102"), "Refactoring", "Martin Fowler", 1),
        new Book(Guid.Parse("00000000-0000-0000-0000-000000000103"), "Design Patterns", "Erich Gamma", 3)
    ];

    public Book? GetById(Guid id) => _books.FirstOrDefault(x => x.Id == id);

    public IReadOnlyCollection<Book> GetAll() => _books;
}
